Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -284,6 +284,39 @@ def check_service_command(formula)
     "Service command does not exist" unless File.exist?(formula.service.command.first)
   end
 
+  def check_cpuid_instruction(formula)
+    return unless formula.prefix.directory?
+    # TODO: add methods to `utils/ast` to allow checking for method use
+    return unless formula.path.read.include? "ENV.runtime_cpu_detection"
+    # Checking for `cpuid` only makes sense on Intel:
+    # https://en.wikipedia.org/wiki/CPUID
+    return unless Hardware::CPU.intel?
+
+    # macOS `objdump` is a bit slow, so we prioritise llvm's `llvm-objdump` (~5.7x faster)
+    # or binutils' `objdump` (~1.8x faster) if they are installed.
+    objdump   = Formula["llvm"].opt_bin/"llvm-objdump" if Formula["llvm"].any_version_installed?
+    objdump ||= Formula["binutils"].opt_bin/"objdump" if Formula["binutils"].any_version_installed?
+    objdump ||= which("objdump")
+    objdump ||= which("objdump", ENV["HOMEBREW_PATH"])
+    objdump ||= begin
+      # If the system provides no `objdump`, install binutils instead of llvm since
+      # binutils is smaller and has fewer dependencies.
+      ohai "Installing `binutils` for `cpuid` instruction check..."
+      safe_system HOMEBREW_BREW_FILE, "install", "binutils"
+      Formula["binutils"].opt_bin/"objdump"
+    end
+
+    keg = Keg.new(formula.prefix)
+    has_cpuid_instruction = false
+    keg.binary_executable_or_library_files.each do |file|
```

Comment:
```suggestion
    keg.binary_executable_or_library_files.any? do |file|
```

instead of the `break`